### ISSUE TITLE
az acr login with subject name and issuer authentication error

### ISSUE THREAD

**Issue by danashviro** [Azure/azure-cli#14072]
# az acr login with subject name and issuer authentication error

I'm trying to authenticate to Azure container registry with the cli. I'm using subject name and issuer authentication, and authenticating to azure with service principal.

that's what I do to login to azure:
`az login --service-principal --use-cert-sn-issuer -u {app id} -p {certificate .pem file} --tenant {tenant id}`

and it worked, I got this response:
```
  {                                                           
    "cloudName": "AzureCloud",                                
    "homeTenantId": {tenant id},
    "id": "” 
    "isDefault": true,                                        
    "managedByTenants": [                                     
      {                                                       
        "tenantId": “”
      }                                                       
    ],                                                        
    "name": "subscription name",                               
    "state": "Enabled",                                       
    "tenantId": , 
    "user": {                                                 
      "name": {app id},         
      "type": "servicePrincipal",                             
      "useCertSNIssuerAuth": true                             
    }                                                         
  }             
```
 and then I try to login to the acr with:
`az acr login --name {acr name}`

it fails with:
```
Get Token request returned http error: 401 and server response: {"error":"invalid_client","error_description":"AADSTS700027: Client assertion contains an invalid signature. [Reason - The key was not found., Thumbprint of key used by client: '{certificate thumbprint}', Please visit 'https://developer.microsoft.com/en-us/graph/graph-explorer' and query for 'https://graph.microsoft.com/beta/applications/{app id}' to see configured keys]\r\nTrace ID: 4546d682-75b6-4a8b-9c03-11f4821c5f00\r\nCorrelation ID: 02e4c324-25a9-47ff-a5ba-afdeec0ce5ec\r\nTimestamp: 2020-06-23 08:12:00Z","error_codes":[700027],"timestamp":"2020-06-23 08:12:00Z","trace_id":"4546d682-75b6-4a8b-9c03-11f4821c5f00","correlation_id":"02e4c324-25a9-47ff-a5ba-afdeec0ce5ec","error_uri":"https://login.microsoftonline.com/error?code=700027"}
Traceback (most recent call last):
pip-install-0u_uiwkq\knack\knack\cli.py, ln 206, in invoke
...
```

## To Reproduce:

- `az login --service-principal --use-cert-sn-issuer -u {app id} -p {certificate .pem file} --tenant {tenant id}`
- `az acr login --name {}`

## Expected Behavior
be logged in to the container registry

## Environment Summary
```
Windows-10-10.0.18362-SP0
Python 3.6.6
Shell: cmd.exe

azure-cli 2.8.0

```

**Comment by yungezz** [comment]
routing to appropriate team

**Comment by toddysm** [comment]
@danashviro would you mind updating the Azure CLI. The latest version is 2.8.0 and you seem to be running an outdated version. Please let us know if the issue persists after the upgrade.

**Comment by danashviro** [comment]
> @danashviro would you mind updating the Azure CLI. The latest version is 2.8.0 and you seem to be running an outdated version. Please let us know if the issue persists after the upgrade.

I upgraded to 2.8.0 and the issue persists

**Comment by toddysm** [comment]
@danashviro I just tested this with 2.8.0 on both Windows and Mac and it works without issues. Are you sure that the registry is in the subscription that is set in `az login`? You may need to set the proper subscription. Can you try:
```
az account list
az account set [subscription-where-registry-is]
```

**Comment by yugangw-msft** [comment]
Chances are token from sn+issuer flow might not be supported by ACR. Let me confirm.

**Comment by yugangw-msft** [comment]
@danashviro, if you can provide registry name and the time when this happens, we can cross check the server log for root cause. I can be reached at yugangw at microsoft dot com

**Comment by danashviro** [comment]
@yugangw-msft, I sent you an email with the details

**Comment by yugangw-msft** [comment]
@danashviro, I am fixing this for the upcoming release

**Comment by ohadschn** [comment]
Wait, `az acr login --password` accepts a PEM file???
Please add this to the docs: https://docs.microsoft.com/en-us/cli/azure/acr?view=azure-cli-latest#az_acr_login

**Comment by ohadschn** [comment]
@yugangw-msft is this included in 2.12.0?
